### PR TITLE
Explain when client-side constants in offline-first apps get updated

### PR DESCRIPTION
**Pull request by omnajjar**

### CODE REVIEW

**Comment by omnajjar** [comment]
Needs team review, will remove the draft flag once we're done reviewing it!

File: content/refguide/constants.md [hunk 1]
```diff
@@ -57,5 +57,5 @@ This property defines whether the constant is accessible from client-side expres
 | No *(default)* | The constant will not be sent to the client and will be only accessible from [microflow](microflows) expressions |
 
 {{% alert type="warning" %}}
-When a constant is exposed to the client, Mendix Runtime sends its value to the client so that in addition to microflow expressions, it will also be accessible from nanoflows and page expressions. This means that you should not use sensitive data or secrets such as passwords when a constant is exposed to the client.
+When a constant is exposed to the client, Mendix Runtime sends its value to the client so that in addition to microflow expressions, it will also be accessible from nanoflows and page expressions. For an offline-first application, these constants will be stored on the device upon the initial startup of the application, and will be updated when you have a connection and you startup the application that was redeployed with a new Mendix version or with domain model changes. This means that you should not use sensitive data or secrets such as passwords when a constant is exposed to the client.
```
Comment (ekrem-aktas) [inline content/refguide/constants.md:60]: Constants are also updated when with login / logout, shall we add this as well?

Comment (omnajjar) [reply content/refguide/constants.md]: Done ✔️

File: content/refguide/constants.md [hunk 2]
```diff
@@ -57,5 +57,5 @@ This property defines whether the constant is accessible from client-side expres
 | No *(default)* | The constant will not be sent to the client and will be only accessible from [microflow](microflows) expressions |
 
 {{% alert type="warning" %}}
-When a constant is exposed to the client, Mendix Runtime sends its value to the client so that in addition to microflow expressions, it will also be accessible from nanoflows and page expressions. This means that you should not use sensitive data or secrets such as passwords when a constant is exposed to the client.
+When a constant is exposed to the client, Mendix Runtime sends its value to the client so that in addition to microflow expressions, it will also be accessible from nanoflows and page expressions. For an offline-first application, these constants will be stored on the device upon the initial startup of the application, and will be updated when you login/logout or when you have a connection and you startup the application that was redeployed with a new Mendix version or with domain model changes. This means that you should not use sensitive data or secrets such as passwords when a constant is exposed to the client.
```
Comment (ekrem-aktas) [inline content/refguide/constants.md:60]: Thanks for the summary @omnajjar , shall we add this as a separate paragraph instead? So the section looks like this:

------
...This means that you should not use sensitive data or secrets such as passwords when a constant is exposed to the client.

For a web or hybrid online app, changes to constant's values are reflected when a user refreshes the browser or restarts the app. For an offline-first application or a PWA app, the app stores the constants' values offline use. The app updates the constant's values in the following cases:
* When a user logs in or logs out in the app.
* When you deploy a new version of the app that contains domain model changes used in the offline-first app.

---- 

We should also not forget to remove PWA while merging this doc to 8.

Comment (omnajjar) [reply content/refguide/constants.md:60]: Will remove PWA mentioning from this PR, and will create another PR for 9 including it 👍🏼

Comment (omnajjar) [reply content/refguide/constants.md]: @ekrem-aktas Thanks for the suggestion, it looks better now less compressed in one big sentence!  ✔️

**Review by ekrem-aktas — APPROVED** [review]
LGTM

**Review by ekrem-aktas — CHANGES_REQUESTED** [review]
We should not mention PWA here

**Comment by omnajjar** [comment]
> We should not mention PWA here

Done ✔️

**Review by ekrem-aktas — APPROVED** [review]
LGTM

**Comment by Adam-Dupaski** [comment]
@ConnorLand I think this is for your component, please take a look.

**Comment by ConnorLand** [comment]
> @ConnorLand I think this is for your component, please take a look.

Thank you. Resolving conflict and merging now.

**Comment by ConnorLand** [comment]
This PR was meant for the Constants doc in RefGuide8, and successfully did so before the Mx9 migration. Therefore I've taken the text from Osama and updated the RefGuide8 doc myself. 

I'll shut this PR down now, because it targets the Mx9 doc mistakenly.

**Pull request closed without merging**